pkg/routeagent/controllers/route/route.go

```diff
@@ -419,31 +420,36 @@ func (r *Controller) processNextEndpoint() bool {
 		}
 
 		if endpoint.Spec.ClusterID != r.clusterID {
-			klog.V(6).Infof("setting routes of endpoint object %s", name)
+			klog.V(log.DEBUG_LEVEL).Infof("setting routes for endpoint  %s", name)
 			r.updateIptableRulesForInterclusterTraffic(endpoint.Spec.Subnets)
 			r.endpointWorkqueue.Forget(obj)
 			return nil
 		}
 
+		klog.V(log.DEBUG_LEVEL).Infof("Local Cluster Gateway Node IP is %s", endpoint.Spec.PrivateIP)
+
 		hostname, err := os.Hostname()
 		if err != nil {
 			klog.Fatalf("unable to determine hostname: %v", err)
 		}
 
-		klog.V(6).Infof("Local Cluster Gateway Node IP is %s", endpoint.Spec.PrivateIP)
-
 		// If the endpoint hostname matches with our hostname, it implies we are on gateway node
 		if endpoint.Spec.Hostname == hostname {
-			r.cleanRoutes()
+			klog.Infof("This has become an active gateway node (IP %s)", endpoint.Spec.PrivateIP)
+
 			r.gwVxLanMutex.Lock()
 			defer r.gwVxLanMutex.Unlock()
 
 			r.isGatewayNode = true
+			klog.Infof("Creating vxlan interface: %s", VxLANIface)
 			err = r.createVxLANInterface(VxInterfaceGateway, nil)
 			if err != nil {
 				klog.Fatalf("Unable to create VxLAN interface on GatewayNode (%s): %v", hostname, err)
 			}
-			klog.V(6).Infof("not reconciling routes because we appear to be the gateway host")
+
+			r.cleanRoutes()
```

Comment:
Any specific reason @mangelajo for moving this API below createVxLANInterface?

Reply:
I assume to move it inside the critical section. Same in `handleRemovedEndpoint`. I was thinking the same thing to be safe.

Reply:
Als, because you can't clean routes if vxlan interface does not exist, that resulted always in a logging error "cannot find vx-submariner" which seems like we are doing something wrong, although we aren't

Reply:
It was really missing in that case, but after the work-queue related patches by tom I'm not seeing that.